producers/prometheus/prometheus_test.go

```diff
@@ -37,58 +41,185 @@ func TestNew(t *testing.T) {
 	})
 }
 
-func TestRun(t *testing.T) {
-	Convey("Should create a Prometheus server on the configured port", t, func() {
-		// Ensure the environment is clean
-		if err := os.Unsetenv("LISTEN_PID"); err != nil {
-			panic(err)
-		}
-		if err := os.Unsetenv("LISTEN_FDS"); err != nil {
-			panic(err)
-		}
+func TestPrometheusProducer(t *testing.T) {
+	type testCase struct {
+		name   string
+		input  []producers.MetricsMessage
+		output map[string]*dto.MetricFamily
+	}
 
-		port, err := getEphemeralPort()
-		if err != nil {
-			panic(err)
-		}
+	now := time.Now().UTC()
+	testCases := []testCase{
+		// Simple case. One metric with one tag.
+		testCase{
+			name: "simple case",
+			input: []producers.MetricsMessage{
+				producers.MetricsMessage{
+					Name: "some-message",
+					Datapoints: []producers.Datapoint{
+						producers.Datapoint{
+							Name:  "datapoint-one",
+							Value: 123.456,
+							Tags:  map[string]string{"hello": "world"},
+						},
+					},
+					Timestamp: now.Unix(),
+				},
+			},
+			output: map[string]*dto.MetricFamily{
+				"datapoint_one": &dto.MetricFamily{
+					Name: proto.String("datapoint_one"),
+					Help: proto.String("DC/OS Metrics Datapoint"),
+					Type: dto.MetricType_GAUGE.Enum(),
+					Metric: []*dto.Metric{
+						&dto.Metric{
+							Gauge: &dto.Gauge{Value: proto.Float64(123.456)},
+							Label: []*dto.LabelPair{
+								&dto.LabelPair{Name: proto.String("cluster_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("container_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("executor_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_name"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_principal"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_role"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("hello"), Value: proto.String("world")},
+								&dto.LabelPair{Name: proto.String("hostname"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("mesos_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("task_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("task_name"), Value: proto.String("")},
```

Comment:
Do we need all these labels in this test? It's noise

Reply:
I put them here deliberately; I don't think they're noise. They're an explicit declaration of the tags we expect on metrics exposed by the Prometheus producer, which I think is appropriate for a test.